### PR TITLE
introduce PostgresBaseImage

### PR DESCRIPTION
**Pull request by stephane-airbyte**
using enums instead of strings for postgres database creation

### CODE REVIEW

File: airbyte-integrations/connectors/source-postgres/src/testFixtures/java/io/airbyte/integrations/source/postgres/PostgresTestDatabase.java [hunk 1]
```diff
@@ -19,8 +19,25 @@
 public class PostgresTestDatabase extends
     TestDatabase<PostgreSQLContainer<?>, PostgresTestDatabase, PostgresTestDatabase.PostgresConfigBuilder> {
 
-  static public PostgresTestDatabase in(String imageName, String... methods) {
-    final var container = new PostgresContainerFactory().shared(imageName, methods);
+  public static enum BaseImage {
+
+    POSTGRES_16_BULLSEYE("postgres:16-bullseye"),
+    POSTGRES_12_BULLSEYE("postgres:12-bullseye"),
+    POSTGRES_16_ALPINE("postgres:16-alpine"),
```
Comment (postamar) [inline airbyte-integrations/connectors/source-postgres/src/testFixtures/java/io/airbyte/integrations/source/postgres/PostgresTestDatabase.java:26]: not a blocking comment but I'd appreciate it if this stack contained a change where POSTGRES_16_ALPINE is removed in favour of POSTGRES_16_BULLSEYE

File: airbyte-integrations/connectors/source-postgres/src/testFixtures/java/io/airbyte/integrations/source/postgres/PostgresTestDatabase.java [hunk 2]
```diff
@@ -19,8 +19,25 @@
 public class PostgresTestDatabase extends
     TestDatabase<PostgreSQLContainer<?>, PostgresTestDatabase, PostgresTestDatabase.PostgresConfigBuilder> {
 
-  static public PostgresTestDatabase in(String imageName, String... methods) {
-    final var container = new PostgresContainerFactory().shared(imageName, methods);
+  public static enum BaseImage {
+
+    POSTGRES_16_BULLSEYE("postgres:16-bullseye"),
```
Comment (postamar) [inline airbyte-integrations/connectors/source-postgres/src/testFixtures/java/io/airbyte/integrations/source/postgres/PostgresTestDatabase.java:24]: non-blocking request: could we have some javadoc commentary for each image with at least a hyperlink to its dockerhub page? Might be useful to help maintenance.

**Review by postamar — APPROVED** [review]

File: airbyte-integrations/connectors/source-postgres/src/testFixtures/java/io/airbyte/integrations/source/postgres/PostgresTestDatabase.java (hunk 2 above)
Comment (stephane-airbyte) [reply airbyte-integrations/connectors/source-postgres/src/testFixtures/java/io/airbyte/integrations/source/postgres/PostgresTestDatabase.java:24]: there's really very little info in the individual image's dockerhub page. For example, [postgres:12-bullseye]([url](https://hub.docker.com/layers/library/postgres/12-bullseye/images/sha256-e0bc95703b924da949b2eb8b4ed5af8485b1f35b50c04577a0848b3e6a3a3abd?context=explore)) vs [postgres:16-bullseye]([url](https://hub.docker.com/layers/library/postgres/16-bullseye/images/sha256-e07f542bafb5c272aa2d2fc478b942fb33acb5f0db9e8f33bee15cc2340af951?context=explore)). In addition, the URL changes with each release of a new version of the image...

**Comment by stephane-airbyte** [comment]
### Merge activity

* **Dec 7, 2:39 PM**: @@stephane-airbyte started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/32851).

**Pull request merged**